CONTRIBUTING.md

```diff
@@ -0,0 +1,84 @@
+# Contributing
+
+## Getting Started
+
+First, you'll need a couple things installed: Docker and asdf.
+With those, you can install all the other tooling that is necessary.
+Assuming you're on macOS, run the following.
```

Comment:
Does this only support macOS? Can we give instructions for other OSs?

Reply:
The best I can do here is set up a Vagrantfile or make a 100% Docker environment. Since I already have all of CI running in containers, I'm probably halfway to the latter.

Reply:
Do you think it's useful to set up the toolchain natively in Windows, or is it tough enough that it's easier to just do a Vagrantfile?

Reply:
@edgriebel my main issue is that I have no way to test it. Making macOS and Docker versions are something I can do. At least with the Docker version, I can also hit the use case where someone someone wants to make a quick change and not install a bunch of stuff they'll never use again.

Reply:
I decided to defer this to #19.